docs/reference/TensorFlow.md

```diff
@@ -1,405 +1,370 @@
 
-Use the following command:
+While the model in this sample was originally trained to recognise “yes” and “no”, the TensorFlow Lite for Microcontrollers source contains scripts that make it easy to retrain the model to classify any other combination of these words.
 
-```cp ./BUILD/K66F/GCC_ARM/mbed.bin /Volumes/K66F/```
+We are going to use another pre-trained model to recognise “up” and “down”:
 
-Depending on your platform, the exact copy command and paths may vary. When you have copied the file, the LEDs on the board should start flashing, and the board will eventually reboot with the sample program running.
+1. Look at how the TensorFlow model gets converted to the TensorFlow Lite format.
+1. Download a model that has been trained and frozen using TensorFlow.
+1. Convert the TensorFlow Lite model into a C source file.
+1. Modify the code and deploy to the board.
 
-# Test keyword spotting
+<span class="tips">**Tip:** Building TensorFlow and training the model will each take a couple of hours on an average computer. We will not perform this at this stage. If you are interested in the full flow, including the training of the model, refer to the [Supplementary information: model training](https://developer.arm.com/solutions/machine-learning-on-arm/developer-material/how-to-guides/build-arm-cortex-m-voice-assistant-with-google-tensorflow-lite/supplementary-information-model-training) guide.</span>
 
-The program outputs recognition results to its serial port. To see the output of the program, we will need to establish a serial connection with the board at 9600 baud.
+## Background: Converting the trained model
 
-The board’s USB UART shows up as  ```/dev/tty.usbmodemXXXXXXX```.We can use ‘screen’ to access the serial console. Although, ‘screen’ is not installed on Linux by default, you can use apt-get install screen to install the package.
+To get a converted model that can run on the controller itself from the trained model, we need to run a conversion script: the [TensorFlow Lite converter](https://www.tensorflow.org/lite/convert). This tool makes our model as small and efficient as possible, and converts it to a TensorFlow Lite FlatBuffer. To reduce the size of the model, we used a technique called [quantization](https://www.tensorflow.org/lite/performance/post_training_quantization). All weights and activations in the model get converted from 32-bit floating point format to an 8-bit and fixed-point format, as you can see in the following command:
 
-Run the following command in a separate terminal:
+<span class="images">![Convert the model to the TensorFlow Lite format code](https://raw.githubusercontent.com/COTASPAR/K66F/master/images/convert_model.png)</span>
```

Comment:
Can this image be made part of this PR's scope?

Reply:
This should be hosted on Arm's GitHub and link from there.

Reply:
Needs to be updated to a relative link and image part of the PR